### ISSUE TITLE
Min-Vid is not maximized when sending a playlist to the player if previously was minimized

### ISSUE THREAD

**Issue by Softvision-VladBacia** [meandavejustice/min-vid#941]
[Affected versions]:
- Firefox 53.0.3 and up
- Min-Vid 1.0.10

[Affected Platforms]:
- All Windows

[Prerequisites]:
- Latest Min-Vid add-on was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites and navigate to a supported platform.
2. Send a video to the Min-Vid player.
3. Minimize the Min-Vid player.
4. Send a Playlist to Min-Vid player.

[Expected result]:
- The Player is maximized and the Playlist overlay is displayed.

[Actual result]:
- The player remains minimized and the Playlist overlay is truncated.

[Notes]:
- Attached a screen recording of the issue:
![minimized state play playlist](https://user-images.githubusercontent.com/9248513/28883430-132e8f08-77b7-11e7-8602-747ab25f2fea.gif)

**Comment by meandavejustice** [comment]
fixed in https://github.com/meandavejustice/min-vid/commit/65ed04d2858eb47d0885a27a60132be375b41411

**Comment by Softvision-VladBacia** [comment]
Today we've built the Min-Vid extension from the `sdk-to-webextension` branch in order to test the isses that were fixed. With that build we encountered other issues. When testing the above scenario the browser and mini player become unresponsive for several seconds multiple times. It looks as the browser wants to crash and in the console there are multiple errors with `too much recursion` . The tests were performed on Nightly 57. 

- Are there issues with the building process? 
- Do we need to build from another branch?
- Is Nightly affecting the extension?
- Is the extension compatible with Nightly 57?

Here is a video with the behavior:

![ssend playlist to player](https://user-images.githubusercontent.com/9248513/29027821-f32652f4-7b8a-11e7-93ea-a2eaad510a93.gif)
